snap_plugin/v1/plugin.py

```diff
@@ -375,10 +419,14 @@ def _generate_preamble_and_serve(self):
                     "RPCVersion": self.meta.rpc_version,
                     "ConcurrencyCount": self.meta.concurrency_count,
                     "Exclusive": self.meta.exclusive,
-                    "Unsecure": self.meta.unsecure,
                     "CacheTTL": self.meta.cache_ttl,
                     "RoutingStrategy": self.meta.routing_strategy,
+                    "RootCertPaths": self.meta.root_cert_paths,
+                    "CertPath": self.meta.server_cert_path,
+                    "PrivateKeyPath": self.meta.private_key_path,
+                    "Ciphers": self.meta.cipher_suites,
                 },
+                "TLSEnabled": self._args.tls,
```

Comment:
Field `"TLSEnabled"` must on the same level as other TLS-related properties (`"CertPath"`, `"KeyPath"`).
Otherwise framework won't recognize plugin as secure.